iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -621,170 +745,222 @@ private async Task RefreshSasTokenLoopAsync(DateTime refreshesOn, CancellationTo
 
         private async Task VerifyIsOpenAsync(CancellationToken cancellationToken)
         {
-            await _handlerSemaphore.WaitAsync(cancellationToken);
+            if (Logging.IsEnabled)
+                Logging.Enter(this, cancellationToken, nameof(VerifyIsOpenAsync));
 
             try
             {
-                if (!_isOpen)
+                ThrowIfDisposed();
+                await _clientOpenSemaphore.WaitAsync(cancellationToken);
+
+                try
                 {
-                    throw new InvalidOperationException($"The client connection must be opened before operations can begin. Call '{nameof(OpenAsync)}' and try again.");
+                    if (GetClientTransportStatus() != ClientTransportStatus.Open)
```

Comment:
consider checking status, and if opening, then wait the semaphore. That way during happy path, these methods don't wait for the semaphore and slow each other down.